feature/experimental/policy/policy_base/base_vrf_selection_test.go

```diff
@@ -0,0 +1,393 @@
+	niConfPath = dut.Config().NetworkInstance("20")
+	niConf = configNetworkInstance("20", &ateDst2)
+	niConfPath.Replace(t, niConf)
+
+	// Configure default NI and forwarding policy
+	t.Logf("*** Configuring default instance forwarding policy on DUT ...")
+	dutConfPath := dut.Config().NetworkInstance(*deviations.DefaultNetworkInstance)
+	dutConfPath.Type().Replace(t, telemetry.NetworkInstanceTypes_NETWORK_INSTANCE_TYPE_DEFAULT_INSTANCE)
+	policyDutConf := configForwardingPolicy()
+	dutConfPath.PolicyForwarding().Replace(t, policyDutConf)
+}
+
+func configInterfaceDUT(i *telemetry.Interface, me, peer *attrs.Attributes, subintfindex uint32, vlan uint16) *telemetry.Interface {
+	i.Description = ygot.String(me.Desc)
+	i.Type = telemetry.IETFInterfaces_InterfaceType_ethernetCsmacd
+	if *deviations.InterfaceEnabled {
+		i.Enabled = ygot.Bool(true)
+	}
+
+	// Create subinterface
+	s := i.GetOrCreateSubinterface(subintfindex)
+
+	if vlan != 0 {
+		// Add VLANs
+		singletag := s.GetOrCreateVlan().GetOrCreateMatch().GetOrCreateSingleTagged()
+		singletag.VlanId = ygot.Uint16(vlan)
+	}
+	// Add IPv4 stack
+	s4 := s.GetOrCreateIpv4()
+	if *deviations.InterfaceEnabled {
+		s4.Enabled = ygot.Bool(true)
+	}
+	a := s4.GetOrCreateAddress(me.IPv4)
+	a.PrefixLength = ygot.Uint8(plen4)
+
+	// Add IPv6 stack
+	s6 := s.GetOrCreateIpv6()
+	if *deviations.InterfaceEnabled {
+		s6.Enabled = ygot.Bool(true)
+	}
+	s6.GetOrCreateAddress(me.IPv6).PrefixLength = ygot.Uint8(plen6)
+
+	return i
+}
+
+// Configure Network instance on the DUT
+func configNetworkInstance(name string, peer *attrs.Attributes) *telemetry.NetworkInstance {
+	d := &telemetry.Device{}
+	ni := d.GetOrCreateNetworkInstance(name)
+
+	ni.Type = telemetry.NetworkInstanceTypes_NETWORK_INSTANCE_TYPE_L2L3
+	static := ni.GetOrCreateProtocol(telemetry.PolicyTypes_INSTALL_PROTOCOL_TYPE_STATIC, "STATIC")
+	ipv4Nh := static.GetOrCreateStatic("0.0.0.0/0").GetOrCreateNextHop(peer.IPv4)
+	ipv4Nh.NextHop, _ = ipv4Nh.To_NetworkInstance_Protocol_Static_NextHop_NextHop_Union(peer.IPv4)
+	ipv6Nh := static.GetOrCreateStatic("::/0").GetOrCreateNextHop(peer.IPv6)
+	ipv6Nh.NextHop, _ = ipv6Nh.To_NetworkInstance_Protocol_Static_NextHop_NextHop_Union(peer.IPv6)
+	ipv4Nh.Recurse = ygot.Bool(true)
+	ipv6Nh.Recurse = ygot.Bool(true)
+
+	return ni
+}
+
+func configForwardingPolicy() *telemetry.NetworkInstance_PolicyForwarding {
+	d := &telemetry.Device{}
+	ni := d.GetOrCreateNetworkInstance(*deviations.DefaultNetworkInstance)
+	ipv4Address := "0.0.0.0/0"
+	ipv6Address := "::/0"
+	ipv6dest := "2001:db8::1/128"
+
+	// Match policy
+	policyFwding := ni.GetOrCreatePolicyForwarding()
+
+	fwdPolicy1 := policyFwding.GetOrCreatePolicy("match-ipv4")
+	fwdPolicy1.GetOrCreateRule(1).GetOrCreateIpv4().DestinationAddress = ygot.String(ipv4Address)
+	fwdPolicy1.GetOrCreateRule(1).GetOrCreateAction().NetworkInstance = ygot.String("10")
+	fwdPolicy1.GetOrCreateRule(2).GetOrCreateIpv6().DestinationAddress = ygot.String(ipv6dest)
```

Comment:
Rule 1 matches IPv4 (the goal of this policy).

What is going on with rules 2 and 3?  Couldn't you just replace this with a single rule 2 to match ipv6Address (all ipv6) and discard it, and get rid of rule 3?

Reply:
This policy allows IPv4 traffic , added additional rule 3 to discard IPv6. Rule 2 is added to allow IPv6 between ATE and DUT port as rule 3 discards IPv6 packets between ATE and DUT.